engine/executor/dm/unitholder.go

```diff
@@ -16,66 +16,45 @@ package dm
 import (
 	"context"
 	"sync"
-	"time"
 
-	"github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/errors"
 	dmconfig "github.com/pingcap/tiflow/dm/dm/config"
 	"github.com/pingcap/tiflow/dm/dm/pb"
 	"github.com/pingcap/tiflow/dm/dm/unit"
-	"github.com/pingcap/tiflow/dm/dm/worker"
 	"github.com/pingcap/tiflow/dm/dumpling"
 	"github.com/pingcap/tiflow/dm/loader"
-	"github.com/pingcap/tiflow/dm/pkg/backoff"
-	"github.com/pingcap/tiflow/dm/pkg/log"
 	"github.com/pingcap/tiflow/dm/syncer"
 	"github.com/pingcap/tiflow/engine/jobmaster/dm/metadata"
 	"github.com/pingcap/tiflow/engine/lib"
-	"go.uber.org/zap"
 )
 
 // unitHolder hold a unit of DM
 type unitHolder interface {
 	Init(ctx context.Context) error
-	Tick(ctx context.Context) error
-	Stage() metadata.TaskStage
-	Status(ctx context.Context) interface{}
 	Close(ctx context.Context) error
+	Pause(ctx context.Context) error
+	Resume(ctx context.Context) error
+	Stage() (metadata.TaskStage, *pb.ProcessResult)
```

Comment:
maybe another function to return ProcessResult. Stage and Status is separated, so Result can also be separated.

Reply:
I think they should be together, and for finished and error, it should return the result